signalfx-java/src/main/java/com/signalfx/signalflow/ServerSentEventsTransport.java

```diff
@@ -0,0 +1,531 @@
+                }
+
+                return response;
+
+            } catch (IOException ex) {
+                throw new SignalFlowException(
+                        "failed post [ " + httpPost + " ] reason: " + ex.getMessage(), ex);
+            } catch (URISyntaxException ex) {
+                throw new SignalFlowException("failed post reason " + ex.getMessage(), ex);
+            }
+        }
+
+        public void close() throws IOException {
+            client.close();
+        }
+    }
+
+    /**
+     * Computation channel fed from a Server-Sent Events stream.
+     */
+    public static class TransportChannel extends Channel {
+
+        protected static final Logger log = LoggerFactory.getLogger(TransportChannel.class);
+
+        private TransportConnection connection;
+        private CloseableHttpResponse response;
+        private HttpEntity responseHttpEntity;
+        private TransportEventStreamParser streamParser;
+
+        public TransportChannel(final TransportConnection connection,
+                                final CloseableHttpResponse response)
+                throws IOException {
+            super();
+            this.connection = connection;
+            this.response = response;
+            this.responseHttpEntity = response.getEntity();
+            this.streamParser = new TransportEventStreamParser(
+                    this.responseHttpEntity.getContent());
+            this.iterator = this.streamParser;
+
+            if (log.isDebugEnabled()) {
+                log.debug(
+                        "constructed " + this.toString() + " of type " + this.getClass().getName());
+            }
+        }
+
+        public void close() {
+            super.close();
+
+            try {
+                this.response.close();
+            } catch (IOException ex) {
+                log.error("failed to close response", ex);
+            }
+
+            try {
+                this.connection.close();
+            } catch (IOException ex) {
+                log.error("failed to close connection", ex);
+            }
+
+            this.streamParser.close();
+        }
+    }
+
+    public static class TransportEventStreamParser implements Iterator<StreamMessage>, Closeable {
+
+        protected static final Logger log = LoggerFactory
+                .getLogger(TransportEventStreamParser.class);
+
+        private static final String EVENT = "event";
+        private static final String ID = "id";
+        private static final String DATA = "data";
+        private static final String RETRY = "retry";
+        private static final String DEFAULT_EVENT = "message";
+        private static final String EMPTY_STRING = "";
+        private static final Pattern DIGITS_ONLY = Pattern.compile("^[\\d]+$");
+
+        private BufferedReader eventStreamReader;
+        private boolean endOfStreamReached = false;
+
+        private Integer reconnectionTimeout = 1000; // default is 1 second
```

Comment:
Why is this an `Integer` and not an `int`? Also, consider including the unit in the variable name, like `reconnectionTimeoutMs`